akka-docs/src/main/paradox/scala/general/actor-systems.md

```diff
@@ -107,14 +107,153 @@ which is a single point of contention if over-used.
+
+@@snip [BlockingDispatcherSample.scala](../../../../test/scala/docs/actor/BlockingDispatcherSample.scala) { #blocking-main }
+
+Here the app is sending 100 messages to `BlockingFutureActor` and `PrintActor` and large numbers
+of akka.actor.default-dispatcher threads are handling requests. When you run run the above code,
+you will likely to see the entire application gets stuck somewhere like this:
+
+```
+>　PrintActor: 44
+>　PrintActor: 45
+```
+
+(i.e.) (Ignoring `println`'s latency) `PrintActor` is non-blocking but it cannot process all the 100 messages asap.
+
+In the thread state diagrams below the colours have the following meaning:
+
+ * Turquoise - Sleeping state
+ * Orange - Waiting state
+ * Green - Runnable state
+
+The thread information was recorded using the YourKit profiler, however any good JVM profiler 
+has this feature (including the free and bundled with the Oracle JDK VisualVM, as well as Oracle Flight Recorder). 
+
+The orange portion of the thread shows that it is idle. Idle threads are fine -
+they're ready to accept new work. However, large amounts of Turquoise (sleeping) threads are very bad!
+
+![dispatcher-behaviour-on-good-code.png](../../images/dispatcher-behaviour-on-bad-code.png)
+
+The app is exposed to the load of `i: Int` messages which will block these threads. 
+For example "`default-akka.default-dispatcher2,3,4`"
+are going into the blocking state, after having been idle. It can be observed
+that the number of new threads increases, "`default-akka.actor.default-dispatcher 18,19,20,...`" 
+however they go to sleep state immediately, thus wasting the
+resources.
+
+The number of such new threads depends on the default dispatcher configuration,
+but it will likely not exceed 50. Since many requests are being processed, the entire
+thread pool is starved. The blocking operations dominate threads in the dispatcher 
+such that it has no thread available to handle other requests.
+
+In essence, the `Thread.sleep` operation has dominated all threads and caused anything 
+executing on the default dispatcher to starve for resources (including any actor
+that you have not configured an explicit dispatcher for).
+
+## Solution: Dedicated dispatcher for blocking operations
+
+In `application.conf`, the dispatcher dedicated to blocking behaviour should
+be configured as follows:
+
+```conf
+my-blocking-dispatcher {
+  type = Dispatcher
+  executor = "thread-pool-executor"
+  thread-pool-executor {
+    // or in Akka 2.4.2+
+    fixed-pool-size = 16
+  }
+  throughput = 100
+}
+```
+
+There are many dispatcher options available which can be found in @ref[Dispatchers](../dispatchers.md).
+
+Here `thread-pool-executor` is used, which has a hardcoded limit of threads. It keeps a set number of threads
+available that allow for safe isolation of the blocking operations. The size settings should depend on the app's
+functionality and the number of cores the server has.
```

Comment:
The exact size should be fine tuned depending on the workload you're expecting to run on this dispatcher as well as the number of cores of the machine you're running the application on. Usually a small number around the number of cores is a good default to start from.